### PR TITLE
feat: Add multi tenancy to S3

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Update S3 to support multi-tenancy 
* Update unit test 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/objectStorageService/s3DataService.ts [hunk 1]
```diff
@@ -109,6 +129,7 @@ export class S3DataService implements Persistence {
     }
 
     async deleteResource(request: DeleteResourceRequest) {
+        this.assertValidTenancyMode(request.tenantId);
         await this.dbPersistenceService.readResource(request);
         await S3ObjectStorageService.deleteBasedOnPrefix(request.id);
```
Comment (carvantes) [inline src/objectStorageService/s3DataService.ts]: delete should consider the `tenantId`

**Review by carvantes — CHANGES_REQUESTED** [review]
Overall looks good, but I think changes are needed for the delete operation.

File: src/objectStorageService/s3DataService.ts (hunk 1 above)
Comment (Bingjiling) [reply src/objectStorageService/s3DataService.ts]: Good catch! Updating now...

File: src/objectStorageService/s3DataService.test.ts [hunk 2]
```diff
@@ -49,6 +51,131 @@ describe('SUCCESS CASES: Testing create, read, update, delete of resources; vers
...
+                resourceType: 'Binary',
+                contentType: 'application/pdf',
+                presignedPutUrl: 'https://VALID_S3_PUT_URL.com/tenant1/id_1.pdf/VALID_TEMP_CREDENTIAL',
+            },
+        });
+
+        expect(response.resource.data).toBeUndefined();
+        expect(response.resource.content).toBeUndefined();
+        expect(response.resource.id).toBeDefined();
+        expect(response.resource.meta).toBeDefined();
```
Comment (rsmayda) [inline src/objectStorageService/s3DataService.test.ts:78]: Nit: may be able to shorten your tests by use [strictEqual](https://jestjs.io/docs/expect#tostrictequalvalue)

**Review by carvantes — APPROVED** [review]

**Pull request merged**